lib/commands/device/deactivate.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright 2016-2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { flags } from '@oclif/command';
+import type { IArg } from '@oclif/parser/lib/args';
+import Command from '../../command';
+import * as cf from '../../utils/common-flags';
+import { getBalenaSdk, stripIndent } from '../../utils/lazy';
+
+interface FlagsDef {
+	yes: boolean;
```

Comment:
`yes: boolean` is correct I believe, because the absence of a bool flag is interpreted as a `false` value when parsed, it won't ever be undefined.

Reply:
Oh ok! 👍

Reply:
Except that's not what I am observing in a test:
```
	public async run() {
		const { args: params, flags: options } = this.parse<FlagsDef, ArgsDef>(
			DeviceDeactivateCmd,
		);

		console.error(`yes=${options.yes}`);
		return;
...
```
```
$ ./bin/balena-dev device deactivate foo
yes=undefined
```

Reply:
@pdcastro Hmm, the Typescript checking does not predict this - I wonder why?  We should investigate, since all the other commands use non-optional booleans too.  It doesn't seem to affect functionality though thankfully.

Reply:
I think the framework should be presenting this as a normal bool (true = flag present, false = flag not present).  I don't think it makes sense to have 3 possible values.  I'll try to check how oclif is supposed to work in these cases.

Reply:
Maybe the `yes` flag needs to specify `default: false` around here:
https://github.com/balena-io/balena-cli/blob/v12.29.0/lib/utils/common-flags.ts#L60-L63

Reply:
Yeah I did some tests, and that limits it to true/false. So perhaps we should standardise on that?   We could miss it, but there are no bad side effects if we do (and perhaps we could subclass that flags.boolean, with the default set). I don't know why it doesn't default to false by default (I've asked a question on oclif repo).

Reply:
@mbalamat I'd prefer if we use the `default:false` approach, as it keeps casting out of the logic, and better typing in the flags.  Could you please amend the common flags file?